curriculum/challenges/english/08-coding-interview-prep/rosetta-code/self-referential-sequence.english.md

```diff
@@ -0,0 +1,144 @@
+---
+id: 5eb3e4a21f462f409d656c73
+title: Self-referential sequence
+challengeType: 5
+forumTopicId: 385317
+---
+
+## Description
+
+<section id='description'>
+There are several ways to generate a self-referential sequence. One very common one (the <a href="http://rosettacode.org/wiki/Look-and-say sequence" target="_blank">Look-and-say sequence</a>) is to start with a positive integer, then generate the next term by concatenating enumerated groups of adjacent alike digits:
+0, 10, 1110, 3110, 132110, 1113122110, 311311222110 ...
+The terms generated grow in length geometrically and never converge.
+Another way to generate a self-referential sequence is to summarize the previous term.
+Count how many of each alike digit there is, then concatenate the sum and digit for each of the sorted enumerated digits. Note that the first five terms are the same as for the previous sequence.
+0, 10, 1110, 3110, 132110, 13123110, 23124110 ... see <a href="http://rosettacode.org/wiki/oeis:A036058" target="_blank">The On-Line Encyclopedia of Integer Sequences</a>
+Sort the digits largest to smallest. Do not include counts of digits that do not appear in the previous term.
+Depending on the seed value, series generated this way always either converge to a stable value or to a short cyclical pattern. (For our purposes, I'll use converge to mean an element matches a previously seen element.) The sequence shown, with a seed value of 0, converges to a stable value of 1433223110 after 11 iterations. The seed value that converges most quickly is 22. It goes stable after the first element. (The next element is 22, which has been seen before.)
+</section>
```

Comment:
No code blocks here, but the spacing can match the original problem. Let's also remove one unnecessary link and go to the secure Rosetta Code page:

```
<section id='description'>
There are several ways to generate a self-referential sequence. One very common one (the <a href="https://rosettacode.org/wiki/Look-and-say sequence" target="_blank">Look-and-say sequence</a>) is to start with a positive integer, then generate the next term by concatenating enumerated groups of adjacent alike digits:

<pre>0, 10, 1110, 3110, 132110, 1113122110, 311311222110 ...</pre>

The terms generated grow in length geometrically and never converge.

Another way to generate a self-referential sequence is to summarize the previous term.

Count how many of each alike digit there is, then concatenate the sum and digit for each of the sorted enumerated digits. Note that the first five terms are the same as for the previous sequence.

<pre>0, 10, 1110, 3110, 132110, 13123110, 23124110 ...</pre>

Sort the digits largest to smallest. Do not include counts of digits that do not appear in the previous term.

Depending on the seed value, series generated this way always either converge to a stable value or to a short cyclical pattern. (For our purposes, I'll use converge to mean an element matches a previously seen element.) The sequence shown, with a seed value of 0, converges to a stable value of 1433223110 after 11 iterations. The seed value that converges most quickly is 22. It goes stable after the first element. (The next element is 22, which has been seen before.)
</section>
```